### PR TITLE
Allow configurable impressions/metrics slice size

### PR DESCRIPTION
**Pull request by maksimf**

### CODE REVIEW

File: lib/cache/repositories/impressions/memory_repository.rb [hunk 1]
```diff
@@ -8,27 +8,28 @@ def initialize(adapter, config)
             @config = config
           end
 
-          # Store impression data in the selected adapter
+          # Store impression data in the memory adapter if choice
```
Comment (patricioe) [inline lib/cache/repositories/impressions/memory_repository.rb]: > if choice
typo there?

File: lib/cache/repositories/impressions/redis_repository.rb [hunk 2]
```diff
@@ -18,16 +16,17 @@ def add(split_name, data)
             )
           end
 
-          def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
+          # Store impressions in bulk
+          def add_bulk(key, bucketing_key, treatments, time)
             @adapter.redis.pipelined do
-              treatments_labels_change_numbers.each_slice(IMPRESSIONS_SLICE) do |treatments_labels_change_numbers_slice|
-                treatments_labels_change_numbers_slice.each do |split_name, treatment_label_change_number|
+              treatments.each_slice(@config.impressions_slice_size) do |treatments_slice|
```
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: this is when we store in redis/cache right?

how about the slice when we read from redis and post to Split?

File: lib/cache/repositories/impressions/memory_repository.rb (hunk 1 above)
Comment (maksimf) [inline lib/cache/repositories/impressions/memory_repository.rb]: Fixed

File: lib/cache/repositories/impressions/redis_repository.rb (hunk 2 above)
Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: @patricioe we have `@config.impressions_queue_size` for that:
https://github.com/splitio/ruby-client/blob/development/lib/cache/repositories/impressions/redis_repository.rb#L47

Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: then in which case will we have more than 1000 treatments to store? I'm confused.

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: @patricioe I didn't get the question

**Comment by SenhorCastor** [comment]
Closing this, addressed in other PR (207)

**Pull request closed without merging**